smack-integration-test/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatIntegrationTest.java

```diff
@@ -161,4 +695,29 @@ public void roomDestroyed(MultiUserChat alternateMUC, String reason) {
         assertEquals(0, muc.getOccupantsCount());
         assertNull(muc.getNickname());
     }
+
+    /**
+     * Gets a random room name
+     *
+     * @param prefix A prefix to add to the room name for descriptive purposes
+     */
+    private EntityBareJid getRandomRoom(String prefix) throws XmppStringprepException {
+        final String roomNameLocal = prefix + StringUtils.insecureRandomString(6);
+        return JidCreate.entityBareFrom(Localpart.from(roomNameLocal), mucService.getDomain());
+    }
+
+    /**
+     * Destroys a MUC room, ignoring any exceptions.
+     *
+     * @param muc The room to destroy (can be null).
+     */
+    static void tryDestroy(final MultiUserChat muc) {
+        try {
```

Comment:
Try to keep the try-block as small as possible, never swallow exceptions, and never catch `Exception`, instead try to catch as less exception types as possible. This should become

```java
if (muc == null) {
    return;
}
try {
    muc.destory("test fixture teardown", null);
} catch (XMPPErrorException | NoResponseException | NotConnectedException | InterruptedException e) {
    LOGGER.log(Level.FINE, "Exception while destroing MUC", e);
}
```

Reply:
I avoided this pattern, since whilst it should work, it's semi-optimistic teardown code (in that we don't _really_ care if it succeeds or fails, it'd just be better if it did), and as such wanted to keep it clean of logging code and we don't care about what exception it throws, because that isn't the test. 

Is this a pattern you're really keen on starting?
Presently, there's only one other use of LOGGER in the tests.

Reply:
> Is this a pattern you're really keen on starting?

I think I've already use it everywhere in Smack.

> we don't care about what exception it throws, because that isn't the test.

Right now we don't care, hence my example put the log level at FINE. You can even make it FINER if you want. The point is, if we silently swallow exceptions, then we never have an option to observe those without an debugger. But if each and every exception is feed into the logging API, JUL in this case, then we can configure JUL so that it will show each and every exception if we ever need to. And by default, simply not show those exceptions: It would be the same as an empty `catch` block.

Reply:
I think my point is that these are tests, rather than regular code, and as such have a key objective to ensure the reader is aware of what's going on. Abstractions should be clean, flow should be linear and non-branching, and the intent always surfaced and clear and obvious. (Sure, some of these arguments could be made for "regular" code too)

Catching any exception is a indication to the reader that you don't care if this succeeds or fails.

Conversely, adding a logger is to say "someone could need to know about this", but in this instance they don't. Given the stuff in the first paragraph, tests shouldn't ever require debugging. 

I can though imagine a scenario where you're using these tests for granular observability of another system. That feels like a separate use case, where you might want more significant logging than just this one line, and are likely to simply fork the code in a throwaway branch with a plethora of log lines, be they LOGGER, System.out or other.

Reply:
> tests shouldn't ever require debugging.
But they do: Tests are code, code requires debugging sometimes

Reply:
To be debuggable there should be way more log lines throughout this test. I worry that adopting this as a pattern throughout future tests too might put off future contributions (although won't affect me - I have a wonderful employer paying me to do this!)

I maintain that quietly ignoring exceptions that aren't part of the important flow is a common pattern where IO/networking is involved
e.g. https://github.com/apache/commons-io/blob/commons-io-2.1/src/main/java/org/apache/commons/io/IOUtils.java#L311

Reply:
Or indeed https://github.com/igniterealtime/Smack/blob/master/smack-bosh/src/main/java/org/jivesoftware/smack/bosh/XMPPBOSHConnection.java#L344

Reply:
Maybe we could move debugging to a new issue in the tracker? I'd be interested to have a crack at it, and see what could be done to make tests both highly debuggable and highly readable at the same time. Perhaps there's a whole new pattern in test design to had from this.